torch/lib/c10d/FileStore.cpp

```diff
@@ -310,9 +311,15 @@ std::vector<uint8_t> FileStore::compareSet(
   // it might be outdated
   pos_ = refresh(file, pos_, cache_);
   if (cache_.count(regKey) == 0) {
+    // if the key does not exist and current value is empty then set it
+    if (currentValue.empty()) {
```

Comment:
Nit: Right now you have two places in the function where you write a new key/value pair to the file; one here at line 315 and the other at line 323.You can most likely rearrange the `if` conditions and consolidate them into one single place.

Reply:
Thanks! Updated